pkgs/applications/networking/mumble/default.nix

```
@@ -1,111 +1,173 @@
-      ++ (overrides.buildInputs or [ ]);
-
-    cmakeFlags = [
-      "-D g15=OFF"
-      "-D CMAKE_CXX_STANDARD=17" # protobuf >22 requires C++ 17
-    ] ++ (overrides.configureFlags or [ ]);
-
-    preConfigure = ''
-       patchShebangs scripts
-    '';
-
-    passthru.tests.connectivity = nixosTests.mumble;
-
-    meta = with lib; {
-      description = "Low-latency, high quality voice chat software";
-      homepage = "https://mumble.info";
-      license = licenses.bsd3;
-      maintainers = with maintainers; [ felixsinger lilacious ];
-      platforms = platforms.linux;
-    };
-  });
-
-  client = source: generic {
-    type = "mumble";
-
-    nativeBuildInputs = [ qt5.qttools ];
-    buildInputs = [ flac libogg libopus libsndfile libvorbis qt5.qtsvg rnnoise speex ]
-      ++ lib.optional (!jackSupport) alsa-lib
-      ++ lib.optional jackSupport libjack2
-      ++ lib.optional speechdSupport speechd-minimal
-      ++ lib.optional pulseSupport libpulseaudio
-      ++ lib.optional pipewireSupport pipewire;
-
-    configureFlags = [
-      "-D server=OFF"
-      "-D bundled-celt=ON"
-      "-D bundled-opus=OFF"
-      "-D bundled-speex=OFF"
-      "-D bundle-qt-translations=OFF"
-      "-D update=OFF"
-      "-D overlay-xcompile=OFF"
-      "-D oss=OFF"
-      "-D warnings-as-errors=OFF" # conversion error workaround
-    ] ++ lib.optional (!speechdSupport) "-D speechd=OFF"
-      ++ lib.optional (!pulseSupport) "-D pulseaudio=OFF"
-      ++ lib.optional (!pipewireSupport) "-D pipewire=OFF"
-      ++ lib.optional jackSupport "-D alsa=OFF -D jackaudio=ON";
-
-    env.NIX_CFLAGS_COMPILE = lib.optionalString speechdSupport "-I${speechd-minimal}/include/speech-dispatcher";
-
-    postFixup = ''
-      wrapProgram $out/bin/mumble \
-        --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath (lib.optional pulseSupport libpulseaudio ++ lib.optional pipewireSupport pipewire)}"
-    '';
-  } source;
-
-  server = source: generic {
-    type = "murmur";
-
-    configureFlags = [
-      "-D client=OFF"
-    ] ++ lib.optional (!iceSupport) "-D ice=OFF"
-      ++ lib.optionals iceSupport [
-        "-D Ice_HOME=${lib.getDev zeroc-ice};${lib.getLib zeroc-ice}"
-        "-D CMAKE_PREFIX_PATH=${lib.getDev zeroc-ice};${lib.getLib zeroc-ice}"
-        "-D Ice_SLICE_DIR=${lib.getDev zeroc-ice}/share/ice/slice"
+  generic =
+    overrides: source:
+    (overrides.stdenv or stdenv).mkDerivation (
+      source
+      // overrides
+      // {
+        pname = overrides.type;
+        version = source.version;
+
+        nativeBuildInputs = [
+          cmake
+          pkg-config
+          python3
+          qt5.wrapQtAppsHook
+          qt5.qttools
+        ] ++ (overrides.nativeBuildInputs or [ ]);
+
+        buildInputs = [
+          avahi
+          boost
+          poco
+          protobuf
+        ] ++ (overrides.buildInputs or [ ]);
+
+        cmakeFlags = [
+          "-D g15=OFF"
+          "-D CMAKE_CXX_STANDARD=17" # protobuf >22 requires C++ 17
```

There are helpers available that make setting `cmakeFlags` more "semantic" :)
```suggestion
          (lib.cmakeBool "g15" false)
          (lib.cmakeFeature "CMAKE_CXX_STANDARD" "17") # protobuf >22 requires C++ 17
```

---

applied